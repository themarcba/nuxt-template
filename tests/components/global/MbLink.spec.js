import {} from 'jest';
// Doc: https://vue-test-utils.vuejs.org/api/
import { createLocalVue, config, shallowMount } from '@vue/test-utils';
import MbLink from '@/components/global/MbLink.vue';
import NuxtStubs from '@/tests/nuxt-stubs';

const localVue = createLocalVue();

// Doc: https://vue-test-utils.vuejs.org/api/#config
config.showDeprecationWarnings = true;

describe('MbLink.vue', () => {
  const wrapper = shallowMount(MbLink, {
    localVue,
    stubs: NuxtStubs
  });

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it('matches the snapshot', async () => {
    expect(wrapper).toMatchSnapshot();

    await wrapper.setProps({ boxed: true });
    expect(wrapper).toMatchSnapshot();

    await wrapper.setProps({ inverted: true });
    expect(wrapper).toMatchSnapshot();

    await wrapper.setProps({ success: true, danger: false, warning: false, info: false });
    expect(wrapper).toMatchSnapshot();

    await wrapper.setProps({ success: false, danger: true, warning: false, info: false });
    expect(wrapper).toMatchSnapshot();

    await wrapper.setProps({ success: false, danger: false, warning: true, info: false });
    expect(wrapper).toMatchSnapshot();

    await wrapper.setProps({ success: false, danger: false, warning: false, info: true });
    expect(wrapper).toMatchSnapshot();

    await wrapper.setProps({ href: 'https://google.com' });
    expect(wrapper).toMatchSnapshot();

    await wrapper.setProps({ to: { path: '/', query: { id: '123' } } });
    expect(wrapper).toMatchSnapshot();
  });
});
