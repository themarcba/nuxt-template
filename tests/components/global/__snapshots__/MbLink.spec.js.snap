// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MbLink.vue matches the snapshot 1`] = `<button class="text-accent border-accent inline-block font-bold border-b-2 transition-border duration-300 ease-linear hover:border-transparent"></button>`;

exports[`MbLink.vue matches the snapshot 2`] = `<button class="text-accent border-accent inline-block font-bold border-b-2 transition-border duration-300 ease-linear hover:border-transparent my-2 mx-4"></button>`;

exports[`MbLink.vue matches the snapshot 3`] = `<button class="text-accent hover:border-accent inline-block font-bold border-b-2 transition-border duration-300 ease-linear border-transparent my-2 mx-4"></button>`;

exports[`MbLink.vue matches the snapshot 4`] = `<button class="text-success hover:border-success inline-block font-bold border-b-2 transition-border duration-300 ease-linear border-transparent my-2 mx-4"></button>`;

exports[`MbLink.vue matches the snapshot 5`] = `<button class="text-danger hover:border-danger inline-block font-bold border-b-2 transition-border duration-300 ease-linear border-transparent my-2 mx-4"></button>`;

exports[`MbLink.vue matches the snapshot 6`] = `<button class="text-warning hover:border-warning inline-block font-bold border-b-2 transition-border duration-300 ease-linear border-transparent my-2 mx-4"></button>`;

exports[`MbLink.vue matches the snapshot 7`] = `<button class="text-info hover:border-info inline-block font-bold border-b-2 transition-border duration-300 ease-linear border-transparent my-2 mx-4"></button>`;

exports[`MbLink.vue matches the snapshot 8`] = `<a href="https://google.com" class="text-info hover:border-info inline-block font-bold border-b-2 transition-border duration-300 ease-linear border-transparent my-2 mx-4"></a>`;

exports[`MbLink.vue matches the snapshot 9`] = `<a class="text-info hover:border-info inline-block font-bold border-b-2 transition-border duration-300 ease-linear border-transparent my-2 mx-4"></a>`;
